### PR TITLE
fix build with disallowed aliases

### PR DESCRIPTION
**Pull request by volth**
###### Motivation for this change

What was missed in https://github.com/NixOS/nixpkgs/pull/43692

@matthewbauer

### CODE REVIEW

**Comment by volth** [comment]
@GrahamcOfBorg build hydra

**Comment by volth** [comment]
BTW, I think the cases like ```pkgs.gsettings-desktop-schemas``` and ```pkgs.gnome3.gsettings-desktop-schemas``` are aliases too, and with ```allowAliases=false``` only one of them should remain

**Comment by peti** [comment]
@matthewbauer, yes, I would assume that almost all those out-dated names come from https://github.com/NixOS/cabal2nix/blob/master/src/Distribution/Nixpkgs/Haskell/FromCabal/Name.hs.

We can probably throw those obsolete aliases out there. This *might* break our ability to re-generate the package set in `release-18.03`, but it probably won't and even if it does then the issue would be easy to detect and quick to fix. Arguably, that name mapping should live in a data file that's checked into Nixpkgs anyway instead if being hard-coded in the source code.

**Comment by peti** [comment]
I fixed a couple of name mappings (but not all) in https://github.com/NixOS/cabal2nix/commit/a1690c5dfc539dbd8ad2eb8e7474de342144a87b.

**Comment by volth** [comment]
I have no rights to resolve the merge conflict on own PR :(

File: pkgs/misc/vim-plugins/default.nix [hunk 1]
```diff
@@ -3280,7 +3280,7 @@ let
 
   };
 
-} // lib.optionalAttrs (config.allowAliases or true) (with self; {
+} // lib.optionalAttrs (config.allowAliases or false) (with self; {
```
Comment (matthewbauer) [inline pkgs/misc/vim-plugins/default.nix]: We definitely want this to default to true still. Users can set it to false in config.nix.

Comment (volth) [reply pkgs/misc/vim-plugins/default.nix]: Feel free to revert.

It was mainly to ensure the evaluation is done with ```allowAliases=false``` as  #43692 missed so many replacement yet successfully evaluate.

Although defaulting to false might have sense, putting ```allowAliases``` in a row with ```allowUnfree``` and ```allowBroken``` (they default to false as being "bad", but everyone set them to true in confiig.nix)

Comment (volth) [reply pkgs/misc/vim-plugins/default.nix]: I cannot do the fix (nor resolve the merge conflict), as the branch became read-only
Perhaps due to the auto-request for review (```"volth requested review from edolstra, Ericson2314, nbp and peti as code owners 10 hours ago"```) it is frozen until they all will review

Comment (volth) [reply pkgs/misc/vim-plugins/default.nix]: Ok, managed to rebase
I will commit ```s/config.allowAliases or false/config.allowAliases or true/g``` after ofborg eval

**Comment by peti** [comment]
@volth, strangely enough, I also have a mapping at https://github.com/NixOS/cabal2nix/blob/master/src/Distribution/Nixpkgs/Haskell/FromCabal/Name.hs#L146 that says:

    libNixName "udev" = return "systemd";

Does that sound right?

**Comment by volth** [comment]
yes, you are right, ```udev``` is then aliased to ```systemd```
so there should be
```udev``` -> ```systemd``` 
and
```libudev``` -> ```systemd```

**Pull request merged**